lib/glific/tickets.ex

```diff
@@ -166,4 +171,72 @@ defmodule Glific.Tickets do
         {context, Messages.create_temp_message(context.organization_id, "Failure")}
     end
   end
+
+  @doc """
+  Return the count of support ticket, using the same filter as list_supporting
+  """
+  @spec fetch_support_tickets(map()) :: String.t()
+  def fetch_support_tickets(args) do
+    start_time = DateTime.new!(args.filter.start_date, @beginning_of_day, "Etc/UTC")
+    end_time = DateTime.new!(args.filter.end_date, @end_of_day, "Etc/UTC")
+
+    Ticket
+    |> where([m], m.inserted_at >= ^start_time)
+    |> where([m], m.inserted_at <= ^end_time)
+    |> Repo.all(skip_organization_id: true)
+    |> convert_to_csv_string()
+  end
+
+  @default_headers "body,status,topic,inserted_at,opened_by,assigned_to\n"
+  @minimal_map [
+    :body,
+    :status,
+    :topic,
+    :inserted_at,
+    :contact_id,
+    :user_id
+  ]
+
+  @doc false
+  @spec convert_to_csv_string([Ticket.t()]) :: String.t()
+  def convert_to_csv_string(ticket) do
+    ticket
+    |> Enum.reduce(@default_headers, fn ticket, acc ->
+      acc <> minimal_map(ticket) <> "\n"
+    end)
+  end
+
+  @spec minimal_map(Ticket.t()) :: String.t()
+  defp minimal_map(ticket) do
+    ticket
+    |> Map.take(@minimal_map)
+    |> convert_time()
+    |> IO.inspect()
+    |> parse_delimiter(:body)
+    |> Map.values()
+    |> Enum.reduce("", fn key, acc ->
+      acc <> if is_binary(key), do: "#{key},", else: "#{inspect(key)},"
+    end)
+  end
+
+  @doc false
+  @spec parse_delimiter(map(), atom()) :: map()
+  def parse_delimiter(data, key) do
+    data
+    |> Map.get(key)
+    |> then(&put_in(data[key], "\"#{&1}\""))
+  end
+
+  @spec convert_time(map()) :: map()
+  defp convert_time(ticket) do
+    contact_name = Repo.fetch(Contact, ticket.contact_id) |> elem(1) |> Map.get(:name)
+    user_name = Repo.fetch(User, ticket.user_id) |> elem(1) |> Map.get(:name)
```

Comment:
I think we are doing more than just convert time in this function so name accordingly